src/main/java/com/exasol/smalljsonfilesfixture/S3TestSetupLambdaController.java

```diff
@@ -0,0 +1,255 @@
+package com.exasol.smalljsonfilesfixture;
+
+import static java.util.logging.Level.INFO;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Logger;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import jakarta.json.*;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient;
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.iam.IamClient;
+import software.amazon.awssdk.services.iam.model.Policy;
+import software.amazon.awssdk.services.iam.model.Role;
+import software.amazon.awssdk.services.lambda.LambdaAsyncClient;
+import software.amazon.awssdk.services.lambda.LambdaClient;
+import software.amazon.awssdk.services.lambda.model.*;
+import software.amazon.awssdk.services.lambda.model.Runtime;
+import software.amazon.awssdk.services.sts.StsClient;
+
+/**
+ * This class deploys a AWS lambda function that can create and delete a test fixture with many small JSON files on S3.
+ */
+public class S3TestSetupLambdaController implements AutoCloseable {
+    private static final String LAMBDA_FUNCTION_NAME = "create-json-files";
+    private static final String ROLE_NAME = LAMBDA_FUNCTION_NAME + "-role";
+    private static final String POLICY_NAME = LAMBDA_FUNCTION_NAME + "-policy";
+    private static final Logger LOGGER = Logger.getLogger(S3TestSetupLambdaController.class.getName());
+    private static final String ACTION_CREATE = "create";
+    private static final String ACTION_DELETE = "delete";
+    private final List<Closeable> createdResources = new ArrayList<>();
+    private final String accountId;
+    private final String bucket;
+    private final LambdaAsyncClient asyncLambdaClient;
+    private final String owner;
+
+    /**
+     * Create a new instance of {@link S3TestSetupLambdaController}.
+     * 
+     * @param owner               owner referenced in {@code exa:owner} tag
+     * @param bucket              s3-bucket
+     * @param credentialsProvider AWS credentials provider
+     * @throws IOException if something goes wrong
+     */
+    public S3TestSetupLambdaController(final String owner, final String bucket,
+            final AwsCredentialsProvider credentialsProvider) throws IOException {
```

Comment:
In my opinion it's not good when a constructor does a lot of work and throws an exception. Please create a stack factory method that does the work and returns a new instance.

Reply:
Yea, on the other hand I like it that its symmetric: the constructor creates things and the close method deletes them.
With a factory this would be in a different file.